src/modules/request_upload_selfie/app/request_upload_selfie_usecase.py

```diff
@@ -0,0 +1,22 @@
+from src.shared.domain.entities.student import Student
+from src.shared.domain.repositories.selfie_repository_interface import ISelfieRepository
+from src.shared.helpers.errors.domain_errors import EntityError
+
+class RequestUploadSelfieUsecase:
+    def __init__(self, repo:ISelfieRepository):
+        self.repo = repo
+        
+    def __call__(self, ra: str, name: str, email: str) -> dict:
+
+        if not Student.validate_ra(ra):
+            raise EntityError('ra')
+        
+        if not Student.validate_email(email):
+            raise EntityError('email')
+
+        if not Student.validate_name(name):
+            raise EntityError('name')
+
```

Comment:
verificar se tem selfie in review ou approved antes de criar a URL

Reply:
feito!